addons/product/models/product_template.py

```diff
@@ -335,6 +335,8 @@ def copy(self, default=None):
 
     @api.multi
     def name_get(self):
+        # Prefetch the fields used by the `name_get`, so `browse` doesn't fetch other fields
+        self.read(['name', 'default_code'])
```

Comment:
I think is very good to read only 2 fields, but does it change the self.
On below : return [(template.id, '%s%s' % (template.default_code and '[%s] ' % template.default_code or template.name))
                for template in self]  self is the same, Maybe would be better to change there too suggestion: for template in self.read(['name', 'default_code']).

Reply:
The idea is to read all the names and default_codes in one query, this read forces this to happen before the `for` and for all the items in self, so when it is looping in the return it gets those values from the cache.